features/FEATURE_LWIP/lwip-interface/lwip-sys/arch/sys_arch.h

```diff
@@ -47,15 +47,21 @@ typedef struct {
 #define MB_SIZE      8
 
 typedef struct {
-    osMessageQueueId_t   id;
-    osMessageQueueAttr_t attr;
-    char                 queue[MB_SIZE * (sizeof(void*) + sizeof(os_message_t))];
-    os_message_queue_t   data;
+    osEventFlagsId_t   id;
+    osEventFlagsAttr_t attr;
+    os_event_flags_t   data;
+
+    uint8_t     post_idx;
+    uint8_t     fetch_idx;
+    void*       queue[MB_SIZE];
 } sys_mbox_t;
 
+#define SYS_MBOX_FETCH_EVENT 0x1
+#define SYS_MBOX_POST_EVENT  0x2
+
 #define SYS_MBOX_NULL               ((uint32_t) NULL)
-#define sys_mbox_valid(x)           (((*x).id == NULL) ? 0 : 1 )
-#define sys_mbox_set_invalid(x)     ( (*x).id = NULL )
+#define sys_mbox_valid(x)           (((*x).id == NULL) ? 0 : 1)
```

Comment:
Not that you're really editing it, but that's a weird construct. Surely just `((*x).id != NULL)`.